### ISSUE TITLE
Tab order is incorrect on click for dropdowns in forms

### ISSUE THREAD

**Issue by toni-sharpe** [Sage/carbon#1525]
## Problem

In the gif below there are two sequences displaying tab order on a collection of fields on a form.

The first sequence shows the dropdown selection being made with the key-board, where tab order is correct, moving on the next form element.

The second shows the selection made with the mouse; in this example, the tab order is reset. This is not as expected, it should move onto the next form element.

![tab-order](https://user-images.githubusercontent.com/10499070/30061877-0c145d84-9241-11e7-92f4-b20fc0d0001d.gif)

## Solution

No solution is suggested, but possibly focus reverts to the default 0 position which is focused on the container, or the click might be getting triggered on that too when the mouse is used to select the option.

## Related Issues
* should be cross QA'd: https://github.com/Sage/carbon/issues/1472

**Comment by ianoxley** [comment]
This is likely to be fixed as part of #1471

**Comment by ianforster-sage** [comment]
Behaving correctly in v16.7.0
